src/engine/test/assets/decoders/aws-cloudtrail.yml

```diff
@@ -0,0 +1,163 @@
+#     "responseElements": {
+#       "ConsoleLogin": "Failure"
+#     },
+#     "awsRegion": "us-east-1",
+#     "eventName": "ConsoleLogin",
+#     "userIdentity": {
+#       "userName": "some-user-name",
+#       "accessKeyId": "some-access-key",
+#       "type": "IAMUser",
+#       "principalId": "some-principal-id",
+#       "accountId": "0303456"
+#     },
+#     "eventSource": "signin.amazonaws.com",
+#     "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:61.0) Gecko/20100101 Firefox/61.0",
+#     "sourceIPAddress": "7.222.123.101",
+#     "recipientAccountId": "1020304050607080"
+#   },
+#   "integration": "aws"
+# }
+
+parents:
+  - aws
+
+check:
+  - wazuh.modules.aws.source: +s_eq/cloudtrail
+
+normalize:
+  - check:
+      - _json.aws.previousDigestHashAlgorithm: +s_eq/SHA-256
+    map:
+      file.hash.sha256: $_json.aws.previousDigestSignature
+
+  - check:
+      - _json.aws.userIdentity.accountId: +not_exists
+    map:
+      cloud.account.id: $_json.aws.awsAccountId
+
+  - check:
+      - _json.aws.userIdentity.accountId: +exists
+    map:
+      cloud.account.id: $_json.aws.userIdentity.accountId
+
+  - check:
+      - _json.aws.userIdentity.userName: +not_exists
+    map:
+      user.name: $_json.aws.userIdentity.sessionContext.sessionIssuer.userName
+
+  - check:
+      - _json.aws.userIdentity.userName: +exists
+    map:
+      user.name: $_json.aws.userIdentity.userName
+
+  - map:
+      aws.cloudtrail.additional_eventdata: $_json.aws.additionalEventData
+      aws.cloudtrail.api_version: $_json.aws.apiVersion
+      aws.cloudtrail.digest.end_time: $_json.aws.digestEndTime
+      aws.cloudtrail.digest.log_files: $_json.aws.logFiles
+      aws.cloudtrail.digest.newest_event_time: $_json.aws.newestEventTime
+      aws.cloudtrail.digest.oldest_event_time: $_json.aws.oldestEventTime
+      aws.cloudtrail.digest.previous_hash_algorithm: $_json.aws.previousDigestHashAlgorithm
+      aws.cloudtrail.digest.previous_s3_bucket: $_json.aws.previousDigestS3Bucket
+      aws.cloudtrail.digest.public_key_fingerprint: $_json.aws.publicKeyFingerprint
+      aws.cloudtrail.digest.s3_bucket: $_json.aws.digestEndTime
+      aws.cloudtrail.digest.signature_algorithm: $_json.aws.digestSignatureAlgorithm
+      aws.cloudtrail.digest.start_time: $_json.aws.digestStartTime
+      aws.cloudtrail.error_code: $_json.aws.errorCode
+      aws.cloudtrail.error_message: $_json.aws.errorMessage
+      aws.cloudtrail.event_category: $_json.aws.eventCategory
+      aws.cloudtrail.event_type: $_json.aws.eventType
+      aws.cloudtrail.event_version: $_json.aws.eventVersion
+      aws.cloudtrail.flattened.response_elements.ConsoleLogin: $_json.aws.responseElements.ConsoleLogin
+      aws.cloudtrail.insight_details: $_json.aws.insightDetails
+      aws.cloudtrail.management_event: $_json.aws.managementEvent
+      aws.cloudtrail.read_only: $_json.aws.readOnly
+      aws.cloudtrail.recipient_account_id: $_json.aws.recipientAccountId
```

Comment:
When a field is in a different type than expected the assignment is also done, should this be that way? shouldn't we check also data types?